packages/wallets/src/wallets/wallet-factory.ts

```diff
@@ -39,18 +39,38 @@ export class WalletFactory {
         return this.createWallet(args);
     }
 
-    public async getClientSideWallet<C extends Chain>(args: WalletArgsFor<C>): Promise<Wallet<C>> {
-        const existingWallet = await this.apiClient.getWallet(`me:${this.getChainType(args.chain)}:smart`);
-        if ("error" in existingWallet) {
-            throw new WalletNotAvailableError(JSON.stringify(existingWallet));
+    // Client-side
+    public async getWallet<C extends Chain>(args: WalletArgsFor<C>): Promise<Wallet<C>>;
+    // Server-side
+    public async getWallet<C extends Chain>(walletLocator: string, args: WalletArgsFor<C>): Promise<Wallet<C>>;
+    public async getWallet<C extends Chain>(
+        argsOrLocator: string | WalletArgsFor<C>,
+        maybeArgs?: WalletArgsFor<C>
+    ): Promise<Wallet<C>> {
+        let walletLocator: string;
+        let args: WalletArgsFor<C>;
+
+        if (typeof argsOrLocator === "string") {
+            if (!this.apiClient.isServerSide) {
+                throw new WalletCreationError(
+                    "getWallet with walletLocator is not supported on client side, use getOrCreateWallet instead"
+                );
+            }
+            if (maybeArgs == null) {
+                throw new WalletCreationError("Args parameter is required when walletLocator is provided");
```

Comment:
lets not call it args in an error message so devs can understand what they need to pass in